python/tank/authentication/shotgun_shared/saml2_sso.py

```diff
@@ -0,0 +1,576 @@
+        elif url.startswith(session.host + URL_LOGIN_PATH):
+            # If we are being redirected to the login page, then SSO is not
+            # enabled on that site.
+            session.error = HTTP_AUTHENTICATE_SSO_NOT_UPPORTED
+
+        if session.error:
+            # If there are any errors, we exit by force-closing the dialog.
+            log.error("==- on_http_response_finished: %s - %s" % (url, error))
+            self._dialog.reject()
+
+    def is_handling_event(self):
+        """
+        Called to know if an event is currently being handled.
+        """
+        # log.debug("==- is_handling_event")
+        return self._event_data is not None
+
+    def handle_event(self, event_data):
+        """
+        Called to start the handling of an event.
+        """
+        log.debug("==- handle_event")
+
+        if not self.is_handling_event():
+            self._event_data = event_data
+
+            # At this point, we want to stop any background session renewal, as
+            # it may impede with our new sesion login.
+            self.stop_session_renewal()
+
+            self.start_new_session(event_data)
+        else:
+            log.error("Calling handle_event while event %s is currently being handled" % self._event_data["event"])
+
+    def resolve_event(self, end_session=False):
+        """
+        Called to return the results of the event.
+        """
+        log.debug("==- resolve_event")
+
+        if self.is_handling_event():
+            # rvc.sendInternalEvent(self._event_data["event"], self._session.assembleSession())
+            if end_session:
+                self.end_current_session()
+            self._event_data = None
+        else:
+            log.error("Called resolve_event when no event is being handled.")
+
+    def get_session_data(self):
+        """Returns the relevant session data for the toolkit."""
+        return (
+            self._session.host,
+            self._session.user_id,
+            self._session.session_id,
+            self._session.cookies,
+            self._session.session_expiration
+        )
+
+    def get_session_error(self):
+        """Returns the the error string of the last failed operation."""
+        res = None
+        if self._session and len(self._session.error) > 0:
+            res = self._session.error
+        return res
+
+    ############################################################################
+    #
+    # QTimer callbacks
+    #
+    ############################################################################
+
+    def on_schedule_sso_session_renewal(self):
+        """
+        Called to trigger the session renewal.
+
+        The session renewal, via the off-screen QWebView, will be done at the
+        next time the application event loop does not have any pending events.
+        """
+        log.debug("==- on_schedule_sso_session_renewal")
```

Comment:
These methods are only used in the context of RV and will be moved in an application-specific wrapper later on.

The log message may be less useful to support people, but they are useful to the developer when the ticket is bumped to them. I see no harm in leaving the log entries here.

Given the simplicity of the methods, the method name is self-explanatory. The flow of events tells the story.